controllers/hosting/hostingdeployment_controller.go

```diff
@@ -468,22 +469,28 @@ func convertTimeOrDefault(time *time.Time, defaultValue *metav1.Time) *metav1.Ti
 	return &converted
 }
 
-// Get the SageMaker Endpoint name given a HostingDeployment.
-func GetSageMakerEndpointName(desiredDeployment hostingv1.HostingDeployment) string {
-	SMMaxLen := 63
-	name := desiredDeployment.ObjectMeta.GetName()
-	uid := strings.Replace(string(desiredDeployment.ObjectMeta.GetUID()), "-", "", -1)
-	smEndpointName := name + "-" + uid
-	if len(smEndpointName) > SMMaxLen {
-		smEndpointName = name[:SMMaxLen-len(uid)-1] + "-" + uid
-	}
-	return smEndpointName
-}
-
 func (r *HostingDeploymentReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&hostingv1.HostingDeployment{}).
 		// Ignore status-only and metadata-only updates
 		WithEventFilter(predicate.GenerationChangedPredicate{}).
 		Complete(r)
 }
+
+// Get the Kubernetes name of the Model/EndpointConfig. This must be idempotent so that future reconciler invocations
+// are able to find the object.
+// Kubernetes resources can have names up to 253 characters long.
+// The characters allowed in names are: digits (0-9), lower case letters (a-z), -, and .
+func GetKubernetesNamespacedName(objectName string, hostingDeployment hostingv1.HostingDeployment) types.NamespacedName {
+	k8sMaxLen := 253
+	uid := hostingDeployment.ObjectMeta.GetUID()
+	generation := strconv.FormatInt(hostingDeployment.ObjectMeta.GetGeneration(), 10)
+	generatedPostfix := generation + "-"
```

Comment:
See comment on line 157 on `common.go`

Reply:
I did this intentionally, check response earlier.